diff --git a/docroot/modules/contrib/commerce/modules/price/src/TwigExtension/PriceTwigExtension.php b/docroot/modules/contrib/commerce/modules/price/src/TwigExtension/PriceTwigExtension.php
index 672a951..1afa9b9 100644
--- a/docroot/modules/contrib/commerce/modules/price/src/TwigExtension/PriceTwigExtension.php
+++ b/docroot/modules/contrib/commerce/modules/price/src/TwigExtension/PriceTwigExtension.php
@@ -47,7 +47,7 @@ public static function formatPrice($price) {
       $number_formatter = \Drupal::service('commerce_price.number_formatter_factory')->createInstance();
       $currency_storage = \Drupal::entityTypeManager()->getStorage('commerce_currency');
       $currency = $currency_storage->load($price['currency_code']);
-      return $number_formatter->formatCurrency($price['number'], $currency);
+      return $number_formatter->formatCurrency(round($price['number'], 2), $currency);
     }
     else {
       throw new \InvalidArgumentException('The "commerce_price_format" filter must be given a price object or an array with "number" and "currency_code" keys.');
