diff --git a/src/Service/SimplesamlphpDrupalAuth.php b/src/Service/SimplesamlphpDrupalAuth.php
index 4258173..ae8dfc0 100644
--- a/src/Service/SimplesamlphpDrupalAuth.php
+++ b/src/Service/SimplesamlphpDrupalAuth.php
@@ -95,7 +95,8 @@ class SimplesamlphpDrupalAuth {
     if (!$account) {
       $account = $this->externalRegister($authname);
     }
-
+    // Cypres Saml Configuration.
+    get_saml_user_attributes($account);
     if ($account) {
       // Determine if roles should be evaluated upon login.
       if ($this->config->get('role.eval_every_time')) {
@@ -135,7 +136,7 @@ class SimplesamlphpDrupalAuth {
 
     // It's possible that a user with their username set to this authname
     // already exists in the Drupal database.
-    $existing_user = $this->entityManager->getStorage('user')->loadByProperties(array('name' => $authname));
+    $existing_user = $this->entityManager->getStorage('user')->loadByProperties(array('mail' => $authname));
     $existing_user = $existing_user ? reset($existing_user) : FALSE;
     if ($existing_user) {
       // If auto-enable SAML is activated, link this user to SAML.
